backend/main.py

```
@@ -42,396 +43,327 @@
+            parsed_label_filters = json.loads(label_filters)
+        
+        return await migration_service.discover_containers(
+            container_identifier=container_identifier,
+            identifier_type=identifier_type,
+            label_filters=parsed_label_filters,
+            source_host=source_host,
+            source_ssh_user=source_ssh_user,
+            source_ssh_port=source_ssh_port
         )
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)